lib/flipper/test/shared_adapter_test.rb

```diff
@@ -0,0 +1,239 @@
+module SharedAdapterTests
+  def setup
+    super
+    @flipper = Flipper.new(@adapter)
+    @actor_class = Struct.new(:flipper_id)
+    @feature = @flipper[:stats]
+    @boolean_gate = @feature.gate(:boolean)
+    @group_gate = @feature.gate(:group)
+    @actor_gate = @feature.gate(:actor)
+    @actors_gate =  @feature.gate(:percentage_of_actors)
+    @time_gate =  @feature.gate(:percentage_of_time)
+    Flipper.register(:admins) do |actor|
+      actor.respond_to?(:admin?) && actor.admin?
+    end
+
+    Flipper.register(:early_access) { |actor|
+      actor.respond_to?(:early_access?) && actor.early_access?
+    }
+  end
+
+  def teardown
+    super
+    Flipper.unregister_groups
+  end
+
+  def test_name_is_a_symbol
+    refute_empty  @adapter.name
+    assert_kind_of Symbol, @adapter.name
+  end
+
+  def test_has_included_the_flipper_adapter_module
+    assert_includes  @adapter.class.ancestors, Flipper::Adapter
+  end
+
+  def test_returns_correct_default_values_for_gates_if_none_are_enabled
+    expected = {
+      :boolean => nil,
+      :groups => Set.new,
+      :actors => Set.new,
+      :percentage_of_actors => nil,
+      :percentage_of_time => nil,
+    }
+    assert_equal expected, @adapter.get(@feature)
+  end
+
+  def test_can_enable_and_get_value_for_boolean_gate
```

Comment:
Why the split on this test and the following one when compared to the shared_adapter_specs.rb? I think of these as more integration type specs so I like to see that each side of the coin works following each other more than that they work just in isolation.